learning/tour-of-beam/frontend/integration_test/tour_page_test.dart

```diff
@@ -0,0 +1,222 @@
+  for (final node in module.nodes) {
+    if (node is UnitModel) {
+      expect(
+        find.descendant(
+          of: find.byType(ContentTreeBuilder),
+          matching: find.text(node.title),
+        ),
+        findsAtLeastNWidgets(1),
+      );
+    }
+    if (node is GroupModel) {
+      await _checkGroup(node, wt);
+    }
+  }
+}
+
+Future<void> _checkGroup(GroupModel group, WidgetTester wt) async {
+  final contentTreeController = getContentTreeController(wt);
+  contentTreeController.expandGroup(group);
+  await wt.pumpAndSettle();
+  for (final node in group.nodes) {
+    expect(
+      find.descendant(
+        of: find.byType(ContentTreeBuilder),
+        matching: find.text(node.title),
+      ),
+      findsAtLeastNWidgets(1),
+    );
+  }
+}
+
+Future<void> _checkUnitContentLoadsProperly(WidgetTester wt) async {
+  final modules = _getModules(wt);
+
+  final unit = modules.expand((m) => m.nodes).whereType<UnitModel>().first;
+
+  await wt.tapAndSettle(
+    find.descendant(
+      of: find.byType(UnitWidget),
+      matching: find.text(unit.title),
+    ),
+  );
+
+  expect(
+    find.descendant(
+      of: find.byType(UnitContentWidget),
+      matching: find.text(unit.title),
+    ),
+    findsOneWidget,
+  );
+}
+
+Future<void> _checkContentTreeScrollsProperly(WidgetTester wt) async {
+  final modules = _getModules(wt);
+  final lastNode = modules.expand((m) => m.nodes).whereType<UnitModel>().last;
+
+  await wt.ensureVisible(
+    find.descendant(
+      of: find.byType(UnitWidget),
+      matching: find.text(lastNode.title),
+      skipOffstage: false,
+    ),
+  );
+  await wt.pumpAndSettle();
+}
+
+Future<void> _checkDisplayingSelectedUnit(WidgetTester wt) async {
+  final controller = getContentTreeController(wt);
+  final selectedUnit = controller.currentNode;
+
+  if (selectedUnit == null) {
+    return;
+  }
+
+  final selectedUnitText = find.descendant(
+    of: find.byType(UnitWidget),
+    matching: find.text(selectedUnit.title),
+    skipOffstage: false,
+  );
+
+  final selectedUnitContainer = find.ancestor(
+    of: selectedUnitText,
+    matching: find.byKey(UnitWidget.containerKey),
+  );
+
+  final context = wt.element(selectedUnitContainer);
+
+  expect(
+    (wt.widget<Container>(selectedUnitContainer).decoration as BoxDecoration?)
+        ?.color,
+    Theme.of(context).selectedRowColor,
+  );
+}
+
+Future<void> _checkResizeUnitContent(WidgetTester wt) async {
+  var dragHandleFinder = find.byKey(TourScreen.dragHandleKey);
+
+  final startHandlePosition = wt.getCenter(dragHandleFinder);
+
+  await wt.drag(dragHandleFinder, const Offset(100, 0));
+  await wt.pumpAndSettle();
+
+  dragHandleFinder = find.byKey(TourScreen.dragHandleKey);
+
+  final movedHandlePosition = wt.getCenter(dragHandleFinder);
+
+  expectSimilar(startHandlePosition.dx, movedHandlePosition.dx - 100);
+}
+
+Future<void> _checkRunCodeWorks(WidgetTester wt) async {
```

Comment:
No, I don't think so. It's very similar to `runCustomJava` method in playground tests, but I think that using extracted method will be weird, because all that we can extract - put code, that prints given string to code field. For clear understanding I think it's better to keep this logic without extracting for a while. Possible we will extract this, when cases will be more than 2.